python/pyspark/worker.py

```diff
@@ -177,8 +180,11 @@ def process():
             process()
     except Exception:
         try:
+            exc_info = traceback.format_exc()
+            if isinstance(exc_info, unicode):
+                exc_info = exc_info.encode('utf-8')
```

Comment:
Let's wait for the resolution of #17267 if you don't mind. I think we should be careful of this change.